src/CoreLib/DSetAction.fs

```diff
@@ -71,26 +75,38 @@ type internal DSetAction() =
                         for p in parameters do
                             x.Param <- p
     /// Hold a Job object
-    member val Job  = null with get, set
+    member x.Job with get() = jobInternal
+                 and set(v: Job) = Volatile.Write( jobLifecycleRef, JobLifeCycleCollectionApp.BeginJob() )
```

Comment:
Agree. Since each DSetAction only host a single job, I will change all Volatile.Write and .Read to normal reference access.